File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.cpp
class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        vector<bool> l(nums.size());
        long long int k=0;
        for (int i =0; i<nums.size();i++){
            k = ((k<<1) | nums[i])%5;
            l[i] =(k==0);
        }
        return l;
    }
};

File: 1071-binary-prefix-divisible-by-5/binary-prefix-divisible-by-5.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        n = len(nums)
        l=[False]*n
        k = 0 
        for i in range(n):
            k = ((k<<1) | nums[i] )%5
            l[i]= (k==0)
        return l
            
